clar2wasm/tests/wasm-generation/main.rs

```diff
@@ -0,0 +1,253 @@
+            Value::Response(ResponseData { committed, data }) => {
+                if *committed {
+                    write!(f, "(ok {})", PropValue(*data.clone()))
+                } else {
+                    write!(f, "(err {})", PropValue(*data.clone()))
+                }
+            }
+            Value::Sequence(SequenceData::List(ListData { data, .. })) => {
+                write!(f, "(list")?;
+                for d in data {
+                    write!(f, " ")?;
+                    write!(f, "{}", PropValue(d.clone()))?;
+                }
+                write!(f, ")")
+            }
+            Value::Tuple(data) => {
+                write!(f, "(tuple")?;
+                for (key, value) in &data.data_map {
+                    write!(f, " ")?;
+                    write!(f, "({} {})", &**key, PropValue(value.clone()))?;
+                }
+                write!(f, ")")
+            }
+            otherwise => write!(f, "{otherwise}"),
+        }
+    }
+}
+
+impl PropValue {
+    pub fn any() -> impl Strategy<Value = Self> {
+        prop_signature().prop_flat_map(prop_value).prop_map_into()
+    }
+
+    pub fn from_type(ty: TypeSignature) -> impl Strategy<Value = Self> {
+        prop_value(ty).prop_map_into()
+    }
+}
+
+fn prop_value(ty: TypeSignature) -> impl Strategy<Value = Value> {
+    match ty {
+        TypeSignature::NoType => unreachable!(),
+        TypeSignature::IntType => int().boxed(),
+        TypeSignature::UIntType => uint().boxed(),
+        TypeSignature::BoolType => bool().boxed(),
+        TypeSignature::OptionalType(ty) => optional(*ty).boxed(),
+        TypeSignature::ResponseType(ok_err) => response(ok_err.0, ok_err.1).boxed(),
+        TypeSignature::SequenceType(SequenceSubtype::BufferType(size)) => {
+            buffer(size.into()).boxed()
+        }
+        TypeSignature::SequenceType(SequenceSubtype::StringType(StringSubtype::ASCII(size))) => {
+            string_ascii(size.into()).boxed()
+        }
+        TypeSignature::SequenceType(SequenceSubtype::ListType(list_type_data)) => {
+            list(list_type_data).boxed()
+        }
+        TypeSignature::TupleType(tuple_ty) => tuple(tuple_ty).boxed(),
+
+        // TODO
+        TypeSignature::PrincipalType => todo!(),
+        TypeSignature::SequenceType(SequenceSubtype::StringType(StringSubtype::UTF8(_))) => todo!(),
+        TypeSignature::CallableType(_) => todo!(),
+        TypeSignature::ListUnionType(_) => todo!(),
+        TypeSignature::TraitReferenceType(_) => todo!(),
+    }
+}
+
+fn int() -> impl Strategy<Value = Value> {
+    any::<i128>().prop_map(Value::Int)
+}
+
+fn uint() -> impl Strategy<Value = Value> {
+    any::<u128>().prop_map(Value::UInt)
+}
+
+fn bool() -> impl Strategy<Value = Value> {
+    any::<bool>().prop_map(Value::Bool)
+}
+
+pub fn string_ascii(size: u32) -> impl Strategy<Value = Value> {
+    let size = size as usize;
+    prop::collection::vec(0x32u8..0x7e, size..=size).prop_map(|bytes| {
```

Comment:
I believe that should be `32` or `0x20` instead of `0x32`.

Reply:
yes, completely.